descartes_light/core/include/descartes_light/types.h

```diff
@@ -12,8 +15,8 @@ using State = Eigen::Matrix<FloatType, Eigen::Dynamic, 1>;
 template <typename FloatType>
 struct StateSample
 {
-  StateSample(const State<FloatType>& state_, FloatType cost_) : state(state_), cost(cost_) {}
-  StateSample(const State<FloatType>& state_) : StateSample(state_, static_cast<FloatType>(0.0)) {}
+  StateSample(State<FloatType> state_, FloatType cost_) : state(std::move(state_)), cost(cost_) {}
+  StateSample(State<FloatType> state_) : StateSample(std::move(state_), static_cast<FloatType>(0.0)) {}
```

Comment:
`State` is just a typedef for an `Eigen::Matrix`, so we probably shouldn't pass by value. We maybe should may a typedef for an `Eigen::Ref` to a `State`

Reply:
Good catch. I updated to leverage Eigen::Ref.